code-style/scss/README.md

```diff
@@ -1,52 +1,257 @@
+
+We namespace to articulate what class names we wrote. As with the [class naming section above](#Class-Naming-Theory), namespacing is a useful tool to describe what kind of class this is and how it functions. Also, like class naming, the more information in the namespace, the easier it is to comprehend for future developers.
+
+A namespace title should consist of more than one character but no more than four characters. One character rarely provides enough information to describe the type of class, and more than four characters cause the namespace to compete with the class name. Namespaces should be quick to comprehend but only act as an aid to define the type of styles that will exist and how they function.
+
+Our namespacing is derived from the ITCSS section the class belongs in. Since settings, tools, generic, and elements are only used in file naming (and usually use the whole title in the file name), we have only defined ITCSS prefixes for the sections used for class naming:
+
+- `.obj-` (objects)
+- `.cmp-` (components)
+- `.util-` (utilities)
+
+It's important to note that these prefixes are only used in _class_ naming. Sometimes we want to prefix a _file_ name with the ITCSS section to communicate what the file includes, and in that case, we would use the entire name of the section.
+
+
+---
+
+## How ITCSS works with BEM
+
+We use ITCSS and BEM together to help us communicate the purpose of a class. ITCSS shows up both in file naming and in class naming, while BEM is only used for class naming.
+
+### ITCSS File Naming
+
+The parts of ITCSS that are used for file naming don't use BEM. An example of this might be color variables. Those belong in the settings section of the inverted triangle, so we would create a file for that called `settings.variables.color.scss` in which we might define some colors:
+
+```scss
+$c-sky-blue: //value for blue
+$c-crimson: //value for red
+```
+
+We don't need BEM for this because we're not defining classes.
+
+### ITCSS and BEM Class Naming
+
+Objects, components, and utilities are the parts of ITCSS that are used in class naming, and they work together with BEM. Let's look at our example from the [BEM](#BEM) section again:
+
+```html
+<section class="hero">
+  <h1 class="hero__title">...</h1>
+  <p class="hero__description">...</p>
+  <button class="hero__button hero__button--dark">...</button>
+</section>
+```
+
+Right now, we're only using BEM here, but we can add ITCSS to better communicate to other developers what this piece of code is (a component), and what it's for (defining classes for the hero block). This example is a hero section, which is a specific part of our site's UI. That means it's a component in ITCSS. For components, we use `cmp-` to prefix class names and denote their purpose.
+
+```html
+<section class="cmp-hero">
+  <h1 class="cmp-hero__title">...</h1>
+  <p class="cmp-hero__description">...</p>
+  <button class="cmp-hero__button cmp-hero__button--dark">...</button>
+</section>
```

Comment:
I'm not a huge fan of this section. BUT if we document it as _an_ option not _the only_ option I'm more in favor.

Reply:
If something is a BEM block, it's going to be a component or an object. So why would I need to also say it's a component? It's like saying `block-block__element`

Reply:
Also, most of our projects don't really use objects. There might be layout stuff in objects, but not much more than that.

Reply:
This has become pretty common from what I have seen. The argument that was made and why I am putting this in here is that the name spacing a component, `cmp-`, is a way to show that is a component *we* made. We will explain that further in the documentation.